go/arrow/array/numericbuilder.gen.go.tmpl

```diff
@@ -184,6 +184,124 @@ func (b *{{.Name}}Builder) newData() (data *Data) {
 	return
 }
 
+func (b *{{.Name}}Builder) AppendValueFromString(s string) error {
+	if s == NullValueStr {
+		b.AppendNull()
+		return nil
+	}
+  {{if or (eq .Name "Date32") -}}
+  	tm, err := time.Parse("2006-01-02", s)
+    if err != nil {
+      b.AppendNull()
+      return err
+    }
+    b.Append(arrow.Date32FromTime(tm))
+  {{else if or (eq .Name "Date64") -}}
+  	tm, err := time.Parse("2006-01-02", s)
+    if err != nil {
+      b.AppendNull()
+      return err
+    }
+    b.Append(arrow.Date64FromTime(tm))  
+  {{else if or (eq .Name "Time32") -}}
+  	val, err := arrow.Time32FromString(s, b.dtype.Unit)
+    if err != nil {
+      b.AppendNull()
+      return err
+    }
+    b.Append(val)
+  {{else if or (eq .Name "Time64") -}}
+    val, err := arrow.Time64FromString(s, b.dtype.Unit)
+    if err != nil {
+      b.AppendNull()
+      return err
+    }
+    b.Append(val)
+  {{else if or (eq .Name "Timestamp") -}}
+  	v, err := arrow.TimestampFromString(s, b.dtype.Unit)
+    if err != nil {
+      b.AppendNull()
+      return err
+    }
+    b.Append(v)
+  {{else if (eq .Name "Duration") -}}	
+    return fmt.Errorf("AppendValueFromString not implemented for Duration")
+  {{else if or (eq .Name "Int8") -}}
+    v, err := strconv.ParseInt(s, 10, 8)
+    if err != nil {
+      b.AppendNull()
+      return err
+    }
+    b.Append(int8(v))
```

Comment:
You can condense all of these to:

```
v, err := strconv.ParseInt(s, 10, {{.Size}})
if err != nil {
    b.AppendNull()
    return err
}
b.Append({{.name}}(v))
```

instead of repeating it